File: java_selenium/ThreadQaTests/src/main/java/Pages/TextBoxPage.java
package Pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import utils.PropertyReader;

import java.time.Duration;

public class TextBoxPage extends BasePage {

    @FindBy(id = "userName")
    private WebElement name;

    @FindBy(id = "userEmail")
    private WebElement email;

    @FindBy(id = "currentAddress")
    private WebElement currentAddress;

    @FindBy(id = "permanentAddress")
    private WebElement permanentAddress;

    @FindBy(id = "submit")
    private WebElement submitButton;

    @FindBy(id = "name")
    private WebElement nameOutput;

    @FindBy(id = "email")
    private WebElement emailOutput;

    @FindBy(xpath = "//p[@id='currentAddress']")
    private WebElement currentAddressOutput;

    @FindBy(xpath = "//p[@id='permanentAddress']")
    private WebElement permanentAddressOutput;

    public TextBoxPage(WebDriver driver) {
        super(driver);
    }

    public void goToTextBoxPage() {
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(15));
        driver.get(PropertyReader.getProperty("base_url"));
        wait.until(ExpectedConditions.presenceOfElementLocated(By.xpath("//h5[text()='Elements']")));
        driver.findElement(By.xpath("//h5[text()='Elements']")).click();
        driver.findElement(By.xpath("//span[text()='Text Box']")).click();
    }

    public void fillFields(String name, String email, String currentAddress, String permanentAddress) {
        this.name.sendKeys(name);
        this.email.sendKeys(email);
        this.currentAddress.sendKeys(currentAddress);
        this.permanentAddress.sendKeys(permanentAddress);
        submitButton.click();
        this.name.clear();
        this.email.clear();
        this.currentAddress.clear();
        this.permanentAddress.clear();
    }


    public boolean confirmRecord(String expectedName, String expectedEmail, String expectedCurrentAddress, String expectedPermanentAddress) {
//        System.out.println("Expected: " + "Name:" + expectedName + " " + "Email:" + expectedEmail + " " + "Current Address :" + "Permananet Address :" + expectedPermanentAddress + " " + expectedPermanentAddress);
//        System.out.println("Received: " + nameOutput.getText() + " " + emailOutput.getText() + " " + currentAddressOutput.getText() + " " + permanentAddressOutput.getText());
        return nameOutput.getText().equals("Name:" + expectedName) &&
                emailOutput.getText().equals("Email:" + expectedEmail) &&
                currentAddressOutput.getText().equals("Current Address :" + expectedCurrentAddress) &&
                permanentAddressOutput.getText().equals("Permananet Address :" + expectedPermanentAddress);
    }
}

File: python_selenium/ThreadQaTests/page/textbox_page.py
from selenium.webdriver.common.by import By

from page.base_page import BasePage


class TextBoxPage(BasePage):

    def __init__(self, driver):
        super().__init__(driver)

    def go_to_text_box_page(self):
        self.driver.get("http://85.192.34.140:8081/")
        self._find_element(By.XPATH, "//h5[text()='Elements']").click()
        self._find_element(By.XPATH, "//span[text()='Text Box']").click()

    def fill_fields(self, name, email, current_address, permanent_address):
        self._find_element(By.ID, "userName").send_keys(name)
        self._find_element(By.ID, "userEmail").send_keys(email)
        self._find_element(By.ID, "currentAddress").send_keys(current_address)
        self._find_element(By.ID, "permanentAddress").send_keys(permanent_address)
        self._find_element(By.ID, "submit").click()
        self._find_element(By.ID, "userName").clear()
        self._find_element(By.ID, "userEmail").clear()
        self._find_element(By.ID, "currentAddress").clear()
        self._find_element(By.ID, "permanentAddress").clear()

    def confirm_record(self, expected_name, expected_email, expected_current_address, expected_permanent_address):
        return (self._find_element(By.ID, "name").text == "Name:" + expected_name and
                self._find_element(By.ID, "email").text == "Email:" + expected_email and
                self._find_element(By.XPATH,
                                   "//p[@id='currentAddress']").text == "Current Address :"
                + expected_current_address and self._find_element(By.XPATH,
                                                                  "//p[@id='permanentAddress']").text
                == "Permananet Address :" + expected_permanent_address)
